### PR TITLE
Add discussion UI

### LINKED ISSUE #26: Create Discussion Screen
Create the Discussion Creation Screen and test it.

It should be a form that asks the user for a non-empty name, an optional description and a list of participants (just UID/name for now).

### PR DESCRIPTION
**Pull request by dtbcds**
# Add Discussion Screen and UI Tests

## Overview
This PR introduces a new `AddDiscussionScreen` in the `ui` module, allowing users to create a discussion by specifying a title, optional description, and selecting members. It also includes a comprehensive suite of UI tests to ensure proper behavior and navigation.

---

### 1. `AddDiscussionScreen.kt`
- **Purpose**: Provides a Compose UI screen for creating discussions.
- **Features**:
  - Input fields for **title** and **description**.
  - **Live search** for adding members (excluding the current user and already-selected members).
  - **Dropdown menu** for displaying search results with avatars.
  - **Selected members list** with removal functionality.
  - **Create Discussion** button (enabled only when a title is entered).
  - **Discard** button and back navigation handling.
- **UI Elements**:
  - `OutlinedTextField` for text inputs.
  - `LazyColumn` for displaying selected members.
  - `Button` and `OutlinedButton` for actions.
- **Helpers**:
  - `fakeSearchAccounts(query: String)` simulates backend account search.

---

### 2. `AddDiscussionScreenTest.kt`
- **Purpose**: Ensures the `AddDiscussionScreen` UI behaves as expected.
- **Tests included**:
  - **Initial state**: empty fields and disabled create button.
  - **Title input**: enabling create button.
  - **Navigation**: back button and discard button call `goBack()`.
  - **Member search**:
    - Displaying search results.
    - Adding members.
    - Clearing search input.
    - Removing selected members.
    - Filtering out current user and already-selected members.
  - **Dropdown behavior**: closes when search query is empty.

- **Tools**:
  - Compose Test Rule (`createComposeRule`).
  - MockK for mocking `NavigationActions` and `FirestoreViewModel`.
  - Coroutine test support for async operations.

---

## Notes
- The current implementation uses a **fake search** for demonstration. This should later be replaced with a proper backend query.
- Navigation after creating a discussion currently calls `goBack()`. Consider navigating directly to the newly created discussion in a future iteration.
- UI follows the app's theme with `AppColors` and `Material3` components.

---

## Testing
- All tests pass on the Compose Test Rule environment.
- Verified:
  - Input fields behave correctly.
  - Search and member selection work as intended.
  - Navigation actions are correctly triggered.

image

### CODE REVIEW

File: app/src/main/java/com/github/meeplemeet/ui/AddDiscussionScreen.kt [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+              /** Buttons section */
+              Column(
+                  modifier = Modifier.fillMaxWidth(),
+                  horizontalAlignment = Alignment.CenterHorizontally) {
+                    Button(
+                        onClick = {
+                          scope.launch {
+                            viewModel.createDiscussion(
+                                title, description, currentUser, *selectedMembers.toTypedArray())
+                            navigation.goBack() // TODO: navigate to new discussion screen
```
Comment (Atomium90) [inline app/src/main/java/com/github/meeplemeet/ui/AddDiscussionScreen.kt]: It would be better to use callbacks (such as `onBack` and `onCreation`) for handling navigation actions instead of directly depending on the `NavigationActions` model. This would make the composable more reusable and testable, as it won’t be tightly coupled to a specific navigation implementation. The parent can inject the appropriate behavior for both the back button and the creation action.

File: app/src/main/java/com/github/meeplemeet/ui/AddDiscussionScreen.kt [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+          /** App bar with title and back button */
+          CenterAlignedTopAppBar(
+              colors =
+                  TopAppBarDefaults.centerAlignedTopAppBarColors(
+                      containerColor = AppColors.primary,
+                      titleContentColor = AppColors.textIcons,
+                      navigationIconContentColor = AppColors.textIcons),
+              title = { Text(text = "Add Discussion") },
+              navigationIcon = {
+                IconButton(onClick = { navigation.goBack() }) {
```
Comment (Atomium90) [inline app/src/main/java/com/github/meeplemeet/ui/AddDiscussionScreen.kt]: Same comment here. Prefer to directly use a callback `onBack` to decuple UI from navigation logic.

File: app/src/androidTest/java/com/github/meeplemeet/ui/AddDiscussionScreenTest.kt
Comment (Atomium90) [inline app/src/androidTest/java/com/github/meeplemeet/ui/AddDiscussionScreenTest.kt:1]: It looks like there’s no test verifying the integration with the `FirestoreViewModel`.
I would suggest adding a test to check that clicking the "Create Discussion" button actually calls `viewModel.createDiscussion` with the expected parameters. This will help ensure that the UI and ViewModel are properly connected and that the action is triggered as intended.

**Review by Atomium90 — CHANGES_REQUESTED** [review]
### 💬 Overall feedback

This is a **well-structured and comprehensive PR**!  
The new `AddDiscussionScreen` is cleanly implemented, with a clear separation of concerns and thorough in-code documentation.  
The UI tests cover a wide range of behaviors and edge cases, which is great for reliability and maintainability.  

---

### 💡 Suggestions for improvement

- **Decouple navigation logic**  
  Consider replacing direct calls to `NavigationActions` with callbacks (e.g., `onBack`, `onCreation`).  
  This would make the composable more **reusable and testable**, since it wouldn’t depend on a specific navigation model.

- **Add a ViewModel integration test**  
  There’s currently no test verifying that clicking the “Create Discussion” button actually calls  
  `viewModel.createDiscussion(...)` with the expected parameters.  
  Adding such a test would strengthen the connection between the UI and the ViewModel.

- **Handle creation errors gracefully**  
  It would be useful to **handle error cases** (e.g., when `createDiscussion` fails).  
  Currently, there’s no user feedback or state management for such situations — for instance, displaying a snackbar or error message could improve the UX and robustness.

---

### ✅ Summary

Overall, this is a **solid addition** to the codebase.  
With a few small refinements — especially around navigation decoupling and error handling — this feature will be very strong and maintainable.

**Review by Atomium90 — APPROVED** [review]
The requested changes have been applied:

- Navigation is now decoupled via callbacks.  
- An integration test checks the ViewModel call.  
- Error handling with a snackbar is implemented.  

Everything looks good — ready to merge! ✅

**Comment by NilSamaha** [comment]
Overall, this is a well-structured and readable implementation. The composable screen is cleanly organized like the Figma, with clear separation of state, layout, and UI logic. The test suite covers a broad set of UI interactions, including state transitions, navigation callbacks, and user input behavior. Great work!

🔍 Missing Test Coverage

There’s no test verifying that clicking “Create Discussion” actually triggers
viewModel.createDiscussion() with the expected parameters. This is ok for now but will be needed later

💡 Minor Suggestions
	•	The fake search function is fine for testing but might be worth replacing with a dependency-injected mock in the future to avoid coupling UI tests to fake logic.
	•	 All other tests (navigation, search filtering, dropdown handling, enabling/disabling buttons) look solid and deterministic.
	
Summary
✔️ Excellent test coverage and clean code structure. Ready to merge.
